federation-js/src/composition/compose.ts

```diff
@@ -154,6 +154,21 @@ export function buildMapsFromServiceList(serviceList: ServiceDefinition[]) {
 
     externalFields.push(...strippedFields);
 
+    // We need to capture @tag directives from the external fields as well
+    for (const {parentTypeName, field} of externalFields) {
```

Comment:
You're appending fields to `externalFields` in each iteration of the serviceList loop, so you're redundantly checking fields you checked in the last iteration. `for (const {parentTypeName, field} of strippedFields) {` would probably work?

Reply:
(or you could iterate over `externalFields` outside of the serviceList loop?)